hibernate-core/src/main/java/org/hibernate/query/Query.java

```diff
@@ -399,6 +409,24 @@ default Query<R> applyLoadGraph(@SuppressWarnings("rawtypes") RootGraph graph) {
 	@Override
 	Query<R> setLockMode(String alias, LockMode lockMode);
 
+	/**
+	 * Apply a timeout to the corresponding database query.
+	 *
+	 * @param timeout The timeout to apply
+	 *
+	 * @return {@code this}, for method chaining
+	 */
+	Query<R> setTimeout(Timeout timeout);
+
+	/**
+	 * Apply a scope to any pessimistic locking applied to the query.
+	 *
+	 * @param lockScope The lock scope to apply
+	 *
+	 * @return {@code this}, for method chaining
+	 */
+	Query<R> setLockScope(PessimisticLockScope lockScope);
```

Comment:
Steve, not sure if you remember https://hibernate.atlassian.net/browse/HHH-18555. 

`PessimisticLockScope` is current ignored, I believe.

Reply:
I do.  But this does not change anything, good or bad.  We will "fix" `PessimisticLockScope` per the Jira you listed.  That has no bearing on this.

Reply:
OK. That's completely fine by me, it just makes raises the priority of that issue.

Reply:
I don't see why this changes anything there ;)  It does not work now in exactly the same ways it did not work before.

Reply:
Fine.